File: Leetcode/203-Remove-Linked-List-Elements/RemoveLinkedListElements.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        if not head:
            return head
        dummy = ListNode(0)
        dummy.next = head
        cur = dummy
        while cur:
            ite = cur.next
            exists = False
            while ite and ite.val == val:
                exists = True
                ite = ite.next
            if exists:
                cur.next = ite
            cur = ite
        return dummy.next

        

File: Leetcode/203-Remove-Linked-List-Elements/RemoveLinkedListElements.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        if (head == NULL) {
            return head;
        }
        ListNode* dummy = new ListNode(0);
        dummy->next = head;
        ListNode* cur = dummy;
        while (cur != NULL) {
            ListNode* ite = cur->next;
            bool exists = false;
            while (ite!=NULL && ite->val == val) {
                exists = true;
                ite=ite->next;
            }
            if (exists) {
                cur->next = ite;
            }
            cur = ite;
        }
        return dummy->next;
    }
};
